### PR TITLE
Final version of the crawler for this sprint.

### PR DESCRIPTION
**Pull request by KhangNguyenNgoc**
- Updated allocation of resources based directory structure of file system (small or large package)
- Added benchmark function to measure the inserting time into tables as well as metadata extracting time from Exiftool

### CODE REVIEW

File: crawler/crawler/connectPG/extract_data.py [hunk 1]
```diff
@@ -142,6 +142,8 @@ def extract_metadata_generic(self, dbID: int) -> str:
 
             elif att_name in ["file_modify_date", "file_access_date", "file_create_date"]:
                 date_output = refineData(self._input).refine_date(att_exiftool)
+                if '0000-00-00' in date_output:
```
Comment (fischerbenjamin) [inline crawler/crawler/connectPG/extract_data.py:145]: This is okay, but probably always a good idea to tag it with a FIXME/TODO.

Comment (ThomasWolter) [reply crawler/crawler/connectPG/extract_data.py:145]: That's true. I forgot to add the tag.

Comment (fischerbenjamin) [reply crawler/crawler/connectPG/extract_data.py:145]: No problem :) Just saw it ;)

**Comment by ThomasWolter** [comment]
I tried out the tree walk and everything seems to work fine. But I will do some code refactoring next week to clean everything up a bit.

File: crawler/crawler/connectPG/extract_data.py (hunk 1 above)
Comment (ThomasWolter) [reply crawler/crawler/connectPG/extract_data.py:146]: I added the tag, so I will resolve the conversation

**Comment by fischerbenjamin** [comment]
I agree on the refactoring. And we should connect it with the API so we have a basic working demo for the midterm release.

**Pull request merged**